teps/0040-ignore-step-errors.md

```diff
@@ -148,10 +158,135 @@ This proposal is limited to a step within a task and does not address `pipelineT
   * Apply a transformation to the test results (e.g. converts them to a certain format such as junit)
   * Upload the results to a central location used by all the teams
 
+## Proposal
+
+Introduce a new field `exitCode` which takes an integer from 0 to 255 as part of the
+[steps](https://github.com/tektoncd/pipeline/docs/tasks.md#defining-steps) definition along with an `image` and `script`.
+
+```go
+type Step struct {
+
+    corev1.Container `json:",inline"`
+
+    // Script is the contents of an executable file to execute.
+    Script string `json:"script,omitempty"`
+
+    // ...
+
+    // desired exit code of the container
+    //exitCode takes only integer in the range 0 - 255
+    //when set, the entrypoint replaces the exit code of the wrapped command with the value specified here
+    ExitCode *int `json:"exitCode,omitempty"`
+}
+```
+
+A `task` author or a `pipeline` author can use `exitCode` to specify their desired exit code for the step. If `exitCode` is
+set, the entrypoint replaces the original exit code of the `script` or the wrapped command with the value specified in
+the `exitCode`.
+
+To ignore a step error, set the `exitCode` to `0`:
+
+```yaml
+steps:
+  - image: docker.io/library/golang:latest
+    name: ignore-unit-test-failure
+    exitCode: 0
```

Comment:
I like the proposal!
From a terminology perspective, something like `overrideExitCode: 0` feels more intuitive :)

Reply:
.. but we are probably using the podspec API

Reply:
thanks @souleb we discussed this in the API WG today and have almost agreed on the syntax:

```
    exit:
        continue:  [ afterSuccess | always ]
```

Please let me know if you have any inputs. You can watch the screen recording [here](https://youtu.be/eUFpk2sBuC4).